cmd/kpod/ps.go

```diff
@@ -0,0 +1,525 @@
+	"strings"
+	"time"
+
+	"github.com/docker/go-units"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/kubernetes-incubator/cri-o/cmd/kpod/formats"
+	"github.com/kubernetes-incubator/cri-o/libkpod"
+	"github.com/kubernetes-incubator/cri-o/oci"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+type psOptions struct {
+	all     bool
+	filter  string
+	format  string
+	last    int
+	latest  bool
+	noTrunc bool
+	quiet   bool
+	size    bool
+	label   string
+}
+
+type psOutputParams struct {
+	ID         string `json:"id"`
+	Image      string `json:"image"`
+	Command    string `json:"command"`
+	CreatedAt  string `json:"created"`
+	RunningFor string `json:"running"`
+	Status     string `json:"status"`
+	Ports      string `json:"ports"`
+	Size       string `json:"size"`
+	Names      string `json:"names"`
+	Labels     string `json:"labels"`
+	Mounts     string `json:"mounts"`
+	exitCode   int32
+}
+
+const runningState = "running"
+
+var (
+	psFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "Show all the containers, default is only running containers",
+		},
+		cli.StringFlag{
+			Name:  "filter, f",
+			Usage: "Filter output based on conditions given",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "Pretty-print containers to JSON or using a Go template",
+		},
+		cli.IntFlag{
+			Name:  "last, n",
+			Usage: "Print the n last created containers (all states)",
+		},
+		cli.BoolFlag{
+			Name:  "latest, l",
+			Usage: "Show the latest container created (all states)",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "Display the extended informations",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Print the numeric IDs only",
+		},
+		cli.BoolFlag{
+			Name:  "size, s",
+			Usage: "Display the total file sizes",
+		},
+	}
+	psDescription = "Prints out information about the containers"
+	psCommand     = cli.Command{
+		Name:        "ps",
+		Usage:       "List containers",
+		Description: psDescription,
+		Flags:       psFlags,
+		Action:      psCmd,
+		ArgsUsage:   "",
+	}
+)
+
+func psCmd(c *cli.Context) error {
+	config, err := getConfig(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get config")
+	}
+	server, err := libkpod.New(config)
+	if err != nil {
+		return errors.Wrapf(err, "error creating server")
+	}
+	if err := server.Update(); err != nil {
+		return errors.Wrapf(err, "error updating list of containers")
+	}
+
+	if len(c.Args()) > 0 {
+		return errors.Errorf("too many argument, ps takes no arguments")
+	}
+
+	format := genPsFormat(c.Bool("quiet"), c.Bool("size"))
+	if c.IsSet("format") {
+		format = c.String("format")
+	}
+	opts := psOptions{
+		all:     c.Bool("all"),
+		filter:  c.String("filter"),
+		format:  format,
+		last:    c.Int("last"),
+		latest:  c.Bool("latest"),
+		noTrunc: c.Bool("no-trunc"),
+		quiet:   c.Bool("quiet"),
+		size:    c.Bool("size"),
+	}
+
```

Comment:
Docker allows you to use all the flags together, it doesn't error out. If `--latest` and `--last` are given together then `--last` gets priority. 
Here is an example
`➜  kpod git:(kpod_ps) ✗ docker ps -a --latest --last 2
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                  PORTS               NAMES
590d2a25db44        fedora              "sleep 10"          4 days ago          Exited (0) 4 days ago                       happy_einstein
f0bb31b28142        fedora              "sleep 10"          4 days ago          Exited (0) 4 days ago                       stupefied_stallman`